components/webrtc/components/webrtc_notification.jsx

```diff
@@ -123,8 +123,12 @@ export default class WebrtcNotification extends React.Component {
     onCancelCall(message) {
         if (message && message.action !== WebrtcActionTypes.CANCEL) {
             return;
-        } else if (message && message.action === WebrtcActionTypes.CANCEL && this.state.userCalling && message.from_user_id !== this.state.userCalling.id) {
-            return;
+        } else if (message && message.action === WebrtcActionTypes.CANCEL && this.state.userCalling) {
+            if (message.from_user_id === UserStore.getCurrentId() && message.calling_user_id === this.state.userCalling.id) {
```

Comment:
I'm lost here as this *if* basically doesn't do anything, can you write this in a different way?

Reply:
Yes those ifs are pretty hard to read. Still the behavior is correct in this way. The newly added if is meant to avoid ending up in the else clause (and thus is a noop). 

I agree that this can be written better at the cost of a larger change set.

Reply:
If I'm understanding this correctly, maybe we could make it clearer by splitting it into two functions like

```javascript
onWebrtcChange(message) {
    if (message && message.action === WebrtcActionTypes.CANCEL) {
        this.onCancelCall(message);
    }
}

onCancelCall(message) {
    if (message.from_user_id !== UserStore.getCurrentId() || message.calling_user_id !== this.state.userCalling.id) {
        // Handled by another session
        return;
    }

    // TODO cancel stuff
}
```

Reply:
Ok i am going to rewrite this. The case which needs to end up at the `cancel stuff` is when its either a valid cancel (existing behavior) or when its a cancel sent by ourselves to ourselves (which sent by the session which accepts the call).